### PR TITLE
Scribble and Canvas stuff...

### PR DESCRIPTION
**Pull request by CandyFace**
Changes:
+ Canvas renderer should never apply bilinear filtering
(smoothTransform)

Fixed:
+ Canvas could get painted even if frame didn’t exist (bitmap).
+ Update was accidentally called recursively… cpu throttle.

Other: 
+ Refactoring

### CODE REVIEW

File: core_lib/canvasrenderer.cpp [hunk 1]
```diff
@@ -80,7 +80,6 @@ void CanvasRenderer::paint( Object* object, int layer, int frame, QRect rect )
     QPainter painter( mCanvas );
 
     painter.setWorldTransform( mViewTransform );
-    painter.setRenderHint( QPainter::SmoothPixmapTransform, mOptions.bAntiAlias );
```
Comment (chchwy) [inline core_lib/canvasrenderer.cpp]: Curious why remove this line? It's an option that you can switch it on and off in Preferences

Comment (CandyFace) [reply core_lib/canvasrenderer.cpp]: Set renderHint with smoothPixmapTransform serves no purpose here, it creates a similar effect to that of mipmaps for textures, making the output look blurry which is not an effect you would want for pixel drawing application. The anti-aliasing checkbox in preferences should only affect the edges of vector lines (which it does)

mipmap

notice the output becoming blurry as soon as i let go of the mouse button.

Comment (chchwy) [reply core_lib/canvasrenderer.cpp]: Alright, you convinced me. 
Going to merge it.

**Comment by chchwy** [comment]
Mmm...the ignore filter doesn't work
https://travis-ci.org/chchwy/pencil2d/jobs/277990045

**Comment by CandyFace** [comment]
Mmm indeed... I wonder if it's related to committing from master rather than branches. You can see mine worked https://travis-ci.org/CandyFace/pencil/jobs/277793235... I'll have a look at both of them later.

**Comment by chchwy** [comment]
It works when I put another commit to it.
Seems like the ignore filter will fail with a merge request.
https://travis-ci.org/chchwy/pencil2d/jobs/278078388

**Comment by chchwy** [comment]
@CandyFace 
I think I found the problem
Please see the log of this build: https://travis-ci.org/chchwy/pencil2d/jobs/278442027

`git diff --name-only origin/master...${TRAVIS_COMMIT}` output nothing.
it means the shell will work on all branches except `master`.

**Comment by CandyFace** [comment]
I think I have a fix very soon, I just need verify the travis build, then i'll make a pr.

**Pull request merged**